pkgs/applications/editors/tijolo/default.nix

```diff
@@ -0,0 +1,58 @@
+{ stdenv
+, lib
+, fetchFromGitHub
+, crystal
+, vte
+, libgit2
+, editorconfig-core-c
+, gtksourceview4
+, wrapGAppsHook
+, desktopToDarwinBundle
+}:
+crystal.buildCrystalPackage rec {
+  pname = "tijolo";
+  version = "0.7.3";
+
+  src = fetchFromGitHub {
+    owner = "hugopl";
+    repo = "tijolo";
+    rev = "v${version}";
+    hash = "sha256-15not/B+O+wIZ/fvLFy26/dyvq0E+bZUeoSZ6HxMMKg=";
+  };
+
+  nativeBuildInputs = [ wrapGAppsHook ]
+    ++ lib.optionals stdenv.isDarwin [ desktopToDarwinBundle ];
+  buildInputs = [ vte libgit2 gtksourceview4 editorconfig-core-c ];
+
+  format = "make";
+  shardsFile = ./tijolo-shards.nix;
+
+  doCheck = false;
```

Comment:
why?

Reply:
Because this test tule is broken in nix:
```
test:
	# Some tests need en_US locale to pass on string to float convertions: "1.23" vs "1,23".
	@if [ "$$(uname -s)" == "Darwin" ]; then\
	  GC_DONT_GC=1 crystal spec;\
	else\
	  GC_DONT_GC=1 LC_ALL=en_US.UTF8 xvfb-run crystal spec;\
	fi
```
`xvfb-run - command not found`
I tried to add `xvfb-run` in `buildInputs` but for some reason it didn't work.

Reply:
try nativeCheckInputs

Reply:
I will try, thank you for your help.

Reply:
I figured out all problems, but now i'm stuck this
```
(crystal-run-spec.tmp:329): Gtk-CRITICAL **: 14:27:19.744: _gtk_css_provider_load_named: assertion '!g_str_equal (name, DEFAULT_THEME_NAME)' failed
Fontconfig error: No writable cache directories
Fontconfig error: No writable cache directories
Fontconfig error: No writable cache directories
Fontconfig error: No writable cache directories
Fontconfig error: No writable cache directories
```
it's still the check phase.
Terminal freezes, nothing happens.

Reply:
I can't help further with this, it tried building the package locally but it takes way too long to build crystal itself.